plugin.py

```diff
@@ -109,10 +109,14 @@ def install_or_update(cls) -> None:
 
     def on_pre_server_command(self, command: Mapping[str, Any], done_callback: Callable[[], None]) -> bool:
 
-        if len(command["arguments"]) == 0:
+        cargo_commands = []
+        for c in command["arguments"]:
+            if c["kind"] == "cargo":
+                cargo_commands.append(c)
```

Comment:
What is `command['command']` in those cases that you want to handle? Aren't those command names specific enough to match against them?

Reply:
Yeah but they have flags and stuff, much better to just use the RUnnables array in the arguments

Reply:
Can you log the communication that triggers those commands and paste it here?
Do those commands not have a common prefix or something?

I'm not sure if we are on the same page. Would you be able to log the relevant server communication and paste here?

Reply:
Are you sure? the `command['command']` is the name of the command, like Sublime Text commands such as `insert`, `append`, `open_url`.

Reply:
I guess you should check for `command['command']` at the start of this method to ensure that we're not accidentally handling an unknown command.

Reply:
Okay right then it's always rust `rust-analyzer.runSingle`